### PR TITLE
 Run Spotbugs.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
We harden our code by running Spotbug.

JIRA issues: DCOS_OSS-5072

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java [hunk 1]
```diff
@@ -45,7 +45,7 @@
 
   private final URL jenkinsUrl;
 
-  private final List<MesosAgentSpecTemplate> mesosAgentSpecTemplates;
+  private final transient List<MesosAgentSpecTemplate> mesosAgentSpecTemplates;
```
Comment (jeschkies) [inline src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java:52]: I'm not really sure this is correct. What do you think, @kvish?

**Comment by jeschkies** [comment]
Jenkins does not support reporting. We have a pending ticket with DevProd.

File: src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java (hunk 1 above)
Comment (kvish) [reply src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java:52]: shouldnt the templates be part of the persistent state of the class and not marked transient?

Comment (jeschkies) [reply src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java:52]: They should but Spotbugs complains about `MesosAgentSpecTemplate` not being serializable. It probably caught the issue we have. I think we should land this and then follow up with [DCOS_OSS-5058](https://jira.mesosphere.com/browse/DCOS_OSS-5058).

**Review by timcharper — APPROVED** [review]

File: src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java (hunk 1 above)
Comment (kvish) [reply src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java:52]: ok fine with doing it in a follow up

**Review by kvish — APPROVED** [review]

**Pull request merged**